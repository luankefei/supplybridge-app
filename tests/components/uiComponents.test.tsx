import React from "react";
import { render, screen } from "@testing-library/react";
import PoweredBy from "components/ui-components/poweredBy";
import {
  SpacingHorizontal,
  SpacingVertical,
} from "components/ui-components/spacer";
// This has to be mocked before running the test
// jest.mock("next/router", () => require("next-router-mock"));

describe("ui-Components", () => {
  it("poweredBy ok -- matches snapshot", () => {
    const { asFragment } = render(<PoweredBy />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("spacerV ok -- matches snapshot", () => {
    const { asFragment } = render(<SpacingVertical space={"10px"} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("spacerH ok -- matches snapshot", () => {
    const { asFragment } = render(<SpacingHorizontal space={"10px"} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
