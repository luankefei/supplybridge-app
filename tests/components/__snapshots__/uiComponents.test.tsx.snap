// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-Components poweredBy ok -- matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-bimbu8"
  >
    <img
      class="sc-aXZVg jPhcGY"
      height="30"
      src="/icons/smart-bridge-ai.svg"
      title=""
      width="24"
    />
    <div
      class="sc-eqUAAy falwWc"
    />
    <div
      class="sc-kAyceB jPGZVz"
      color="#89A8B3"
      font-size="13px"
      font-weight="400"
    >
      Powered by SmartBridge Artificial Intelligence
    </div>
  </div>
</DocumentFragment>
`;

exports[`ui-Components spacerH ok -- matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-eqUAAy haKPhF"
  />
</DocumentFragment>
`;

exports[`ui-Components spacerV ok -- matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gEvEer lajBMu"
  />
</DocumentFragment>
`;
